src/Guzzle6HttpAdapter.php

```diff
@@ -53,56 +63,42 @@ public function sendRequest(RequestInterface $request, array $options = [])
     /**
      * {@inheritdoc}
      */
-    public function sendRequests(array $requests, array $options = [])
+    public function sendRequests(array $requests)
     {
-        $options = $this->buildOptions($options);
+        $poolResult  = Pool::batch($this->client, $requests, ['options' => $this->options]);
+        $batchResult = new BatchResult();
 
-        $results = Pool::batch($this->client, $requests, $options);
-
-        $exceptions = [];
-        $responses = [];
-
-        foreach ($results as $result) {
+        foreach ($poolResult as $index => $result) {
             if ($result instanceof ResponseInterface) {
-                $responses[] = $result;
-            } elseif ($result instanceof RequestException) {
-                $exceptions[] = $this->createException($result);
+                $batchResult = $batchResult->addResponse($requests[$index], $result);
             }
-        }
 
-        if (count($exceptions) > 0) {
-            throw new Exception\MultiHttpAdapterException($exceptions, $responses);
+            if ($result instanceof RequestException) {
+                $batchResult = $batchResult->addException($requests[$index], $this->createException($result));
+            }
         }
 
-        return $results;
-    }
+        if ($batchResult->hasExceptions()) {
+            throw new BatchException($batchResult);
+        }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function getName()
-    {
-        return 'guzzle6';
+        return $batchResult;
     }
 
     /**
-     * Converts a Guzzle exception into an HttpAdapter exception
+     * Converts a Guzzle exception into an Httplug exception
      *
      * @param RequestException $exception
      *
-     * @return Exception\HttpAdapterException
+     * @return HttpException|NetworkException Return an HttpException if response is available, NetworkException otherwise
```

Comment:
In theory, this can be Http\Client\Exception. If we ever change the Exceptions, the method signature remains correct. Doesn't really matter though, since the method is private.